department-of-veterans-affairs/caseflow
# Mkerns/caseflow 199 add warning message

Resolves #{github issue number}

### Description
Added warning message to warn user if the new nod date is a date after the original nod date.

### Acceptance Criteria
- [x] 199.1 Verify that when the New date is later than original NOD date a warning appears, when the user tabs out of calendar selection

- [x] 199.2 Verify the warning reads: "The new NOD date entered is later than the original date. Changing the NOD date to a later date may negatively affect the time it takes for a Veteran to receive a response to their appeal."

- [x] 199.3 Verify that the Submit button is clickable even if the warning condition is met/not met

### Testing Plan
1. Login as an approved user type (like BVAAABSHIRE)
2. Go to Queue and Search cases.
3. Find a case that has an existing NOD (like 123456789)
4. Goto the Case details page, scroll to the "Edit NOD Date" link
5. Click the link to launch the EditNodDate modal.
6. The "Submit" button is disabled upon initial opening
7. Click the date picker and select a new date that is after the original nod date but not after today's date.
8. The alert should trigger.
9. The submit button should be clickable.

### User Facing Changes

 BEFORE|AFTER
 ---|---
![image](https://user-images.githubusercontent.com/69862429/105770783-aa48d900-5f2d-11eb-9bea-c24b81c37b1c.png)|![image](https://user-images.githubusercontent.com/69862429/105770712-956c4580-5f2d-11eb-83f9-12db1632ec41.png)
